utils/generate_task_parameters.py

```diff
@@ -0,0 +1,528 @@
+# coding=utf-8
+# Copyright 2024-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""**Experimental** Utility script to check and update the InferenceClient task methods arguments and docstrings based on the tasks input parameters."""
```

Comment:
```suggestion
"""
Utility script to check and update the InferenceClient task methods arguments and docstrings
based on the tasks input parameters.

What this script does:
- [x] detect missing parameters in method signature
- [x] add missing parameters to methods signature
- [ ] detect outdated parameters in method signature
- [ ] update outdated parameters in method signature

- [x] detect missing parameters in method docstrings
- [x] add missing parameters to methods docstrings
- [ ] detect outdated parameters in method docstrings
- [ ] update outdated parameters in method docstrings

- [ ] detect when parameter not used in method implementation
- [ ] update method implementation when parameter not used

Related resources:
- https://github.com/huggingface/huggingface_hub/issues/2063
- https://github.com/huggingface/huggingface_hub/issues/2557
- https://github.com/huggingface/huggingface_hub/pull/2561
"""
```

Given how complex the script is, let's be precise about what it does / doesn't. Also good to keep it as a to-do list for future improvements.